packages/flake-tracker/package.json

```diff
@@ -18,6 +18,8 @@
     "@octokit/plugin-retry": "^3.0.3",
     "@octokit/rest": "^18.0.0",
     "clone": "^2.1.2",
+    "detox": "^17.13.2",
+    "mocha": "^7.1.1",
```

Comment:
I couldn't remember why but I just tried removing it and I get an error while building android that says it can't find com.wix.detox:

`Could not determine the dependencies of task ':app:preApptestDebugAndroidTestBuild'.
> Could not resolve all task dependencies for configuration ':app:apptestDebugAndroidTestRuntimeClasspath'.
   > Could not find any matches for com.wix:detox:+ as no versions of com.wix:detox are available.
     Searched in the following locations:
       - file:/home/erdal/.m2/repository/com/wix/detox/
       - file:/home/erdal/dev/celo-monorepo/node_modules/react-native/android/com/wix/detox/maven-metadata.xml
       - file:/home/erdal/dev/celo-monorepo/node_modules/jsc-android/dist/com/wix/detox/maven-metadata.xml
       - file:/home/erdal/dev/celo-monorepo/node_modules/detox/Detox-android/com/wix/detox/maven-metadata.xml
       - file:/home/erdal/dev/celo-monorepo/node_modules/@celo/client/build/bin/
       - https://dl.google.com/dl/android/maven2/com/wix/detox/maven-metadata.xml
       - https://jcenter.bintray.com/com/wix/detox/maven-metadata.xml
       - https://www.jitpack.io/com/wix/detox/maven-metadata.xml
     Required by:
         project :app
`

Obviously this is the wrong solution, so trying to find a better one.

Reply:
I think the problem is detox gets installed in packages/mobile/node_modules and our gradle script looks for it in the root node_modules.

Reply:
Ok this is taking me back to a few weeks ago :) When I remove the package from flake-tracker I get errors when running the e2e tests:

`detox[21699] INFO:  [test.js] configuration="android.emu.debug" loglevel="verbose" artifactsLocation="e2e/artifacts" recordLogs="failing" takeScreenshots="all" useCustomLogger=true forceAdbInstall=false DETOX_START_TIMESTAMP=1607626703988 readOnlyEmu=false reportSpecs=true jest --config e2e/config.json --testNamePattern '^((?!:ios:).)*$' --maxWorkers 1 --detectOpenHandles  
 FAIL  e2e/src/DeepLinkSend.spec.js
  ● Test suite failed to run

    Cannot find module 'detox/runners/jest-circus'

      3 |   SpecReporter,
      4 |   WorkerAssignReporter,
    > 5 | } = require('detox/runners/jest-circus')
        |     ^
      6 | const JestFlakeTracker = require('../tracker')
      7 | 
      8 | class FlakeTrackingDetoxEnv extends DetoxCircusEnvironment {

      at Object.<anonymous> (../../flake-tracker/src/jest/environments/detox.js:5:5)

`

Reply:
so I think it is needed there. What probably happens is the package gets hoisted to the root, which is why we import it from there in the gradle build:
`maven {
            url "$rootDir/../../../node_modules/detox/Detox-android"
        }`